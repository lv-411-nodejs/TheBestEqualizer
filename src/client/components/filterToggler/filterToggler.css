.modesElement {
  text-align: center;
  color: var(--mainText-ordinary-white);
}

.labels {
  margin: 0 5px;
}

.switch {
  position: relative;
  display: inline-block;
  width: 46px;
  height: 20px;
  align-self: center;
}

.switch input {
  display: none;
}

.switch input:checked+.slider {
  background-color: #eee;
}

.switch input:checked+.slider:before {
  -webkit-transform: translateX(26px);
  -moz-transform: translateX(26px);
  -ms-transform: translateX(26px);
  -o-transform: translateX(26px);
  transform: translateX(26px);
  background: #2ECC71;
}

.slider {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: table-cell;
  height: 20px;
  margin-top: 3px;
  cursor: pointer;
  -webkit-transition: .4s;
  -moz-transition: .4s;
  -ms-transition: .4s;
  -o-transition: .4s;
  transition: .4s;
  vertical-align: middle;
  border: 1px solid #444;
  -webkit-border-radius: 34px;
  -moz-border-radius: 34px;
  border-radius: 34px;
  background-color: #414543;
  -ms-border-radius: 34px;
  -o-border-radius: 34px;
}

.slider:before {
  position: absolute;
  bottom: 3px;
  left: 2px;
  width: 13px;
  height: 13px;
  content: '';
  -webkit-transition: .4s;
  -moz-transition: .4s;
  -ms-transition: .4s;
  -o-transition: .4s;
  transition: .4s;
  -webkit-border-radius: 50%;
  -moz-border-radius: 50%;
  border-radius: 50%;
  background-color: #fff;
  -ms-border-radius: 50%;
  -o-border-radius: 50%;
}
