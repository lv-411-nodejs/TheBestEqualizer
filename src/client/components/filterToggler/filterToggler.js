import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './filterToggler.css';
import { filterHandler } from '../../store/actions/audioActions';
import { attachFiltersToSource, removeSourceFilters } from '../../helpers/equalizerAuxMethods';

export const FilterToggler = (props) => {
  const {
    audioData: {
      voice,
      sound,
      filtersToggler,
    },
    blocksData,
    filterHandlerAsProp,
  } = props;
  const onToggler = () => {
    filterHandlerAsProp();
    if (filtersToggler) {
      removeSourceFilters(voice, blocksData);
      attachFiltersToSource(sound, blocksData);
    } else {
      removeSourceFilters(sound, blocksData);
      attachFiltersToSource(voice, blocksData);
    }
  };

  return (
    <div className="modesElement">
      <span className="labels">Sound</span>
      <label htmlFor="toggler" className="switch">
        <input id="toggler" type="checkbox" onClick={onToggler} />
        <span className="slider" />
      </label>
      <span className="labels">Voice</span>
    </div>
  );
};

FilterToggler.propTypes = {
  audioData: PropTypes.instanceOf(Object).isRequired,
  blocksData: PropTypes.instanceOf(Array).isRequired,
  voice: PropTypes.instanceOf(Object),
  sound: PropTypes.instanceOf(Object),
  filterHandlerAsProp: PropTypes.func,
};

const mapStateToProps = state => ({
  audioData: state.audioData,
  blocksData: state.blocksData,
});

export default connect(mapStateToProps, { filterHandlerAsProp: filterHandler })(FilterToggler);
