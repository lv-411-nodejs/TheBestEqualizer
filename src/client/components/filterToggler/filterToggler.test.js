import React from 'react';
import {
    shallow,
    render,
    mount
} from 'enzyme';
import { FilterToggler } from './filterToggler';
// import configureMockStore from "redux-mock-store";


describe('Filter Toggler', () => {
    const props = {
        audioData: {
            onToggle: false,
            sound: {},
            voice: {},
        },
        blocksData: [],
    };
    
    // const mockStore = configureMockStore();
    // let store;
    
    it('should render filterToggler element', () => {
        // store = mockStore({audioData: {
        //     onToggle: false,
        //     sound: {},
        //     voice: {},
        // },
        // blocksData: [],});
        const nextProps = {
            ...props,
        };
        
        const wrapper = shallow(<FilterToggler {...nextProps} />);
        // expect(wrapper.find('.modesElement')).toHaveLength(1);
    });

    it('should render filterToggler properly', () => {
        const nextProps = {
            ...props,
        };
        const wrapper = shallow(<FilterToggler {...nextProps} />);
        console.debug(wrapper);
        expect(wrapper).toMatchSnapshot();
    });

    it('should invoke onToggler function that remove sound and attach voice filters', () => {
        const nextProps = {
            ...props,
        };
    });

});