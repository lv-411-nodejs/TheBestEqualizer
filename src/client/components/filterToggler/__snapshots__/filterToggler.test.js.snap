// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Toggler should invoke onToggler function that remove sound and attach voice filters 1`] = `ShallowWrapper {}`;

exports[`Filter Toggler should render filterToggler properly 1`] = `ShallowWrapper {}`;
